link_shortener/commands/create.py

```diff
@@ -5,23 +5,15 @@
 import hashlib
 import os
 
-from sqlalchemy import and_
-
-from link_shortener.core.exceptions import DuplicateActiveLinkForbidden
+from link_shortener.core.validation import endpoint_duplicity_check
 from link_shortener.models import links, salts
 
 
 async def create_link(request, data):
     async with request.app.engine.acquire() as conn:
         trans = await conn.begin()
-        query = await conn.execute(links.select().where(and_(
-            links.columns['endpoint'] == data['endpoint'],
-            links.columns['is_active'].is_(True)
-        )))
-        link_data = await query.fetchone()
-        if link_data:
-            await trans.close()
-            raise DuplicateActiveLinkForbidden
+
+        await endpoint_duplicity_check(conn, trans, data)
```

Comment:
Hele když přesuneš transakci až za tenhle call tak jí nemusíš posílat pak do tý funkce. Jestli to teda půjde i u ostatních endpointů, to se ještě podívám

Reply:
fixnute
7aef7bebd4361c4c9465efdf8bcb823f54f59285